packages/build/src/telemetry/complete.js

```diff
@@ -13,53 +12,63 @@ const { track } = require('./track')
 
 // Send telemetry request when build completes
 const trackBuildComplete = async function ({
+  status,
   commandsCount,
   netlifyConfig,
   durationNs,
   siteInfo,
   telemetry,
-  mode,
   testOpts,
 }) {
   if (!telemetry) {
     return
   }
 
-  const payload = getPayload({ commandsCount, netlifyConfig, durationNs, siteInfo, mode })
+  const payload = getPayload({ status, commandsCount, netlifyConfig, durationNs, siteInfo })
   await track({ payload, testOpts })
 }
 
 // Retrieve telemetry information
-const getPayload = function ({ commandsCount, netlifyConfig, durationNs, siteInfo: { id: siteId }, mode }) {
-  const durationMs = roundTimerToMillisecs(durationNs)
-  const plugins = Object.values(netlifyConfig.plugins).map(getPluginPackageName)
-  return {
-    type: 'track',
-    event: 'netlifyCI:buildComplete',
-    options: {},
-    userId: null,
-    anonymousId: uuidv4(),
-    meta: {
-      timestamp: Date.now(),
-    },
+const getPayload = function ({ status, commandsCount, netlifyConfig, durationNs, siteInfo }) {
+  const basePayload = {
+    event: 'build:ci_build_process_completed',
+    timestamp: Date.now(),
     properties: {
+      status,
       steps: commandsCount,
-      duration: durationMs,
-      pluginCount: plugins.length,
-      plugins,
-      siteId,
-      isCI,
-      isNetlifyCI: mode === 'buildbot',
       buildVersion: version,
       nodeVersion: nodeVersion.replace('v', ''),
       osPlatform: OS_TYPES[platform],
       osName: osName(),
     },
   }
+
+  return addDuration(addConfigData(addSiteInfoData(basePayload, siteInfo), netlifyConfig), durationNs)
+}
+
+const addSiteInfoData = function (payload, siteInfo = {}) {
+  const userData = siteInfo.user_id ? { userId: siteInfo.user_id } : { anonymousId: uuidv4() }
+  const properties = { properties: { ...payload.properties, siteId: siteInfo.id } }
+  return { ...userData, ...payload, ...properties }
 }
 
-const getPluginPackageName = function ({ packageName }) {
-  return packageName
+const addConfigData = function (payload, netlifyConfig = {}) {
+  if (netlifyConfig.plugins === undefined) return payload
```

Comment:
I believe `netlifyConfig.plugins` should never be `undefined` thanks to:

https://github.com/netlify/build/blob/d7d1dba05120016430892fcbdbf7a2db5c80740b/packages/config/src/normalize.js#L7-L16

Instead, it would be normalized to an empty array.

Reply:
I was just acting based on the default condition above `netlifyConfig = {}`, happy to remove this though 👍

Reply:
Oh, right!
We might want to add a comment there so we remember this.